### PR TITLE
docs: Update contributing link

### LINKED ISSUE #14855: Link directly to contributing guide from `index.md`
### Description

The [Contributing Guide section](https://github.com/pola-rs/polars/blob/main/docs/index.md#contributing) in `index.md` on the main page:

image

Brings you [here](https://github.com/pola-rs/polars/blob/main/CONTRIBUTING.md). Is there a reason we have this interim page which doesn't seem to provide any value?

### Link

_No response_

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #14855.

Made a couple minor edits to other references:

* Use `contributing guide` consistently across the documentation.
* Removed `CONTRIBUTING.md` as it is no longer referenced.
* Provided direct link to appropriate section in the `launch.py` exception message.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (mcrumiller) [inline CONTRIBUTING.md:1]: This file is no longer referenced anywhere in the repository, so it has been removed.

Comment (stinodego) [reply CONTRIBUTING.md:1]: This file needs to stay. GitHub auto-detects this and shows it to users in the UI. And as it's a standard, people will look for this file if they want to contribute.

**Review by stinodego — CHANGES_REQUESTED** [review]
Thanks, one comment then this is good to go.

**Comment by mcrumiller** [comment]
@stinodego restored.

**Review by stinodego — APPROVED** [review]
Great!

**Pull request merged**